File: chroma-league/src/test/java/net/booone/chromaleague/hud/animations/LevelUpAnimationTest.java
package net.booone.chromaleague.hud.animations;

import net.booone.chromaleague.GameStateMocks;
import net.booone.chromaleague.hud.AnimationTester;
import org.junit.jupiter.api.Test;

import static org.mockito.Mockito.when;

class LevelUpAnimationTest {

    @Test
    void playLevelUpAnimation() {
        // given
        when(new GameStateMocks().activePlayer().level()).thenReturn(1);

        // when
        final LevelUpAnimation levelUpAnimation = new LevelUpAnimation();
        levelUpAnimation.levelUp();

        // then
        new AnimationTester().testAnimation(levelUpAnimation);
    }
}

File: test/hud/animations/LevelUpAnimation.test.ts
import {describe, test} from 'vitest';
import {AnimationTester} from '../../AnimationTester.js';
import {GameStateMocks} from '../../GameStateMocks.js';
import {LevelUpAnimation} from '../../../src/hud/animations/LevelUpAnimation.js';

/**
 * Integration test for level up animation on actual hardware.
 */
describe('LevelUpAnimation Integration', () => {
    test('playLevelUpAnimation', async () => {
        // given
        const mocks = new GameStateMocks();
        mocks.activePlayer().level = 1;

        // when
        const levelUpAnimation = new LevelUpAnimation();
        levelUpAnimation.levelUp();

        // then
        await new AnimationTester().testAnimation(levelUpAnimation);
    }, 30000);
});
